codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/flexiblechecksums/FlexibleChecksumsRequestIntegration.kt

```diff
@@ -50,14 +50,16 @@ private object FlexibleChecksumRequestMiddleware : MiddlewareRenderable {
 
     override val position = MiddlewarePosition.AFTER
 
-    override fun render(ctx: ProtocolGenerator.GenerationContext, writer: SwiftWriter, op: OperationShape, operationStackName: String) {
+    override fun renderMiddlewareInit(
+        ctx: ProtocolGenerator.GenerationContext,
+        writer: SwiftWriter,
+        op: OperationShape
+    ) {
```

Comment:
It's a little roundabout how this works, but to render a `MiddlewareRenderable`, `render` gets called, which by default handles `useInterceptors` itself, and calls `renderMiddlewareInit`. Most middleware initialization works exactly the same whether using interceptors or not, so I added some abstraction to `MiddlewareRenderable` in the smithy-swift PR: https://github.com/smithy-lang/smithy-swift/pull/694/files#diff-60728c54157d6f8ad167a46845e0c2c47bc5abb02644411b20f5e5102a113942R48-R70

Reply:
@milesziemer so does that mean `FlexibleChecksumsRequestIntegration` doesn't need a branch for `useInterceptors` because `render` will be called anyway which already goes down that branch? Additionally, should this be refactored into an interceptor instead of a middleware?

Reply:
> so does that mean FlexibleChecksumsRequestIntegration doesn't need a branch for useInterceptors because render will be called anyway which already goes down that branch?

Yes

> Additionally, should this be refactored into an interceptor instead of a middleware?

If you mean the swift implementation, it is, just in https://github.com/smithy-lang/smithy-swift/pull/694